### PR TITLE
docs: couple of ReferenceGrant cleanups

### PR DESCRIPTION
**Pull request by markmc**
Remove deprecated ReferencePolicy using the hideTypePatterns config for the API docs generator.

ReferencePolicy has been renamed to ReferenceGrant, but we have retained ReferencePolicy as a deprecated type in order to make migration easier.

Keeping ReferencePolicy around in the docs is confusing for readers and probably doesn't provide any value.

Also fix path of ReferenceGrant image.

/kind cleanup

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
This makes sense to me :+1: 

Since it has to do with the visibility of an existing API, I'll wait for another maintainer to give the lgtm.

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**
